src/main/java/de/themorpheus/edu/taskservice/controller/TaskController.java

```diff
@@ -152,8 +154,10 @@ else if (Validation.validateNotNullOrEmpty(dto.getDifficultyNameKey()))
 		return ControllerResult.of(tasks.get(RANDOM.nextInt(tasks.size())).toResponseDTO());
 	}
 
-	public ControllerResult<GetAllTasksResponseDTO> getAllTaskByUserId(UUID authorId) {
-		List<TaskModel> tasks = this.taskRepository.findAllByAuthorId(authorId);
+	public ControllerResult<GetAllTasksResponseDTO> getTaskByUserId(UUID authorId, int skip, int max) {
+		skip = Math.max(skip, 0);
+		max = max > TaskServiceApplication.MAX_TASK_RESULTS || max <= 0 ? TaskServiceApplication.MAX_TASK_RESULTS : max;
```

Comment:
Please, calculate this just one time after server-start... for example: in a static {} block in TaskServiceApplication

Reply:
Or make a util or manager or something - that's silly